ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2ODeepLearning.scala

```diff
@@ -34,19 +32,10 @@ import water.support.ModelSerializationSupport
   *
   * TODO: There are still bunch of parameters defined DeepLearningParameters which need to be ported here
   */
-class H2ODeepLearning(parameters: Option[DeepLearningParameters], override val uid: String)
-                     (implicit h2oContext: H2OContext, sqlContext: SQLContext)
-  extends H2OAlgorithm[DeepLearningParameters, H2OMOJOModel](parameters)
+class H2ODeepLearning(override val uid: String) extends H2OAlgorithm[DeepLearningParameters, H2OMOJOModel]
     with H2ODeepLearningParams {
 
-  def this()(implicit h2oContext: H2OContext, sqlContext: SQLContext) = this(None, Identifiable.randomUID("deeplearning"))
-
-  def this(uid: String, hc: H2OContext, sqlContext: SQLContext) = this(None, uid)(hc, sqlContext)
```

Comment:
Only public constructors needs to be this() and this(uid) all others were internal, no doc mentioned them